detox/local-cli/test.js

```diff
@@ -0,0 +1,298 @@
+    number: true
+  },
+  'device-name': {
+    alias: ['n', 'deviceName'],
+    group: 'Configuration',
+    describe: 'Override the device name specified in a configuration. Useful for running a single build configuration on multiple devices.'
+  }
+};
+
+module.exports.handler = function main(program) {
+  program.artifactsLocation = buildDefaultArtifactsRootDirpath(program.configuration, program.artifactsLocation);
+
+  clearDeviceRegistryLockFile();
+
+  if (!program.configuration) {
+    throw new DetoxConfigError(`Cannot determine which configuration to use.
+    Use --configuration to choose one of the following: ${_.keys(config.configurations).join(', ')}`);
+  }
+
+  if (!config.configurations[program.configuration]) {
+    throw new DetoxConfigError(`Cannot determine configuration '${program.configuration}'.
+      Available configurations: ${_.keys(config.configurations).join(', ')}`);
+  }
+
+  const testFolder = getConfigFor(['file', 'specs'], 'e2e');
+  const runner = getConfigFor(['testRunner'], 'mocha');
+  const runnerConfig = getConfigFor(['runnerConfig'], getDefaultRunnerConfig());
+  const platform = config.configurations[program.configuration].type.split('.')[0];
+
+  if (platform === 'android' && program.workers !== 1) {
+    throw new DetoxConfigError('Can not use -w, --workers. Parallel test execution is only supported on iOS currently');
+  }
+
+  if (typeof program.debugSynchronization === 'boolean') {
+    program.debugSynchronization = 3000;
+  }
+
+  function run() {
+    switch (runner) {
+      case 'mocha':
+        runMocha();
+        break;
+      case 'jest':
+        runJest();
+        break;
+      default:
+        throw new Error(`${runner} is not supported in detox cli tools. You can still run your tests with the runner's own cli tool`);
+    }
+  }
+
+  function collectExtraArgs() {
+    const aliasKeys = Object.values(module.exports.builder).reduce(
+      (carry, item) => [].concat(carry, typeof item.alias === 'string' ? [item.alias] : item.alias),
+      []
+    );
+
+    const knownKeys = [].concat(Object.keys(module.exports.builder), aliasKeys);
+
+    const clone = Object.assign({}, program);
+    knownKeys.forEach(key => delete clone[key]);
+
+    const res = Object.entries(clone)
+      .filter(([key]) => !key.includes('_') && !key.includes('-') && !key.includes('$'))
+      .map(([key, value]) => `--${key} ${value}`)
+      .join(' ');
+
+    return res;
+  }
+
+  function getConfigFor(keys, fallback) {
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i];
+      const keyKebabCase = camelToKebabCase(key);
+      const result = program[key] || config[key] || config[keyKebabCase];
+      if (result) return result;
+    }
+
+    return fallback;
+  }
+
+  function camelToKebabCase(string) {
+    return string.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+  }
+
+  function runMocha() {
+    const loglevel = program.loglevel ? `--loglevel ${program.loglevel}` : '';
```

Comment:
🤷‍♂️ Not that I'm aware of, maybe we should have a map between our flags and mocha or jest flags.

Reply:
Interesting...

Reply:
I still hate the fact that mocha has no mechanism like jest for having environment. This would be way more natural from a jest perspective than the wrapping that we currently do